app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -203,35 +178,108 @@ class FirebaseAuthRepository : AuthRepository {
   private fun reauthenticate(
       user: FirebaseUser,
       password: String,
-      activity: Activity,
+      context: Context,
+      coroutineScope: CoroutineScope,
       onSuccess: () -> Unit,
       onFailure: () -> Unit
   ) {
     val isEmail = isEmailProvider(user)
 
     if (isEmail) {
-      val email = user.email ?: throw Exception("User email is null")
-      val credential: AuthCredential = EmailAuthProvider.getCredential(email, password)
-      user.reauthenticate(credential).addOnCompleteListener { task ->
-        if (task.isSuccessful) {
-          Log.d("DeleteAccount", "User re-authenticated")
-          onSuccess()
-        } else {
-          onFailure()
-        }
-      }
+      reauthenticateWithEmailAndPassword(user, password, onSuccess, onFailure)
     } else {
-      val googleProvider = OAuthProvider.newBuilder(GoogleAuthProvider.PROVIDER_ID)
-      user
-          .startActivityForReauthenticateWithProvider(activity, googleProvider.build())
-          .addOnFailureListener { e ->
-            Log.e("DeleteAccount", "Re-authentication failed: ${e.message}")
-            onFailure()
-          }
-          .addOnSuccessListener { result ->
-            Log.d("DeleteAccount", "Re-authentication successful: $result")
-            onSuccess()
+      reauthenticateWithGoogleSignIn(user, context, coroutineScope, onSuccess, onFailure)
+    }
+  }
+
+  private fun reauthenticateWithEmailAndPassword(
```

Comment:
Nitpick: Could you please include a docstring

Reply:
Sure!